Gfieni2022 patch 1 (#57)
CMU-HKN/CMU-ECE-CS-Guide: electives/18847.md

# 18-847C: Data Center Computing

| Category | Difficulty |
|:-:       | :-:        |
| Reviews  | 5          |
| Project  | 6          |
| Present  | 7          |

> This class gives a introduction to data center computing issues by reading recently published research papers addressing current problems that datacenters face today, as well as through a semester long research project into a data center problem. 

Reviewer: ```suggestion
This class gives an introduction to data center computing issues through reading research papers that address  problems that datacenters face today, as well as working through a semester long research project about a data center problem. 
```

## Paper Reviews

> Each class (meet twice a week), you are required to read and review two reseach papers. During class, these papers are presented and discussed. Sometimes, one of the reseachers who wrote the paper will visit for discussion. The most important part of the reviews is the critical analysis of where they could have improved and your discussion questions. 

Reviewer: Do you need to prepare your discussion questions beforehand? Are you graded on how much you participate in these discussions?

```suggestion
Each class (meets twice a week), you are required to read and review two research papers. During class, these papers are presented and discussed. Sometimes, one of the researchers who wrote the paper will visit for discussion. The most important part of the reviews is the critical analysis of where they could have improved and your discussion questions. 
```

Author: See commit https://github.com/gfieni2022/CMU-ECE-CS-Guide/commit/c2eddb433ab1014933531ef80057d0447d8dc86c

## Paper Presentations

> You are required to prepare 2-3 paper presentations throughout the semester. These presentations are 10-15 minutes long and can be relatively informal. Each class, two of these presentations occur, each one focusing on one of the papers for the class. These often take long to prepare, contributing to a higher number of hours spent on the class for those weeks. 

Reviewer: Why do these presentations take a long time to prepare? What are ways to help people get unstuck while they are making these presentations?

Author: It depends on how familiar you are with the material and how good of a presentation you want to make it.

Author: See commit https://github.com/gfieni2022/CMU-ECE-CS-Guide/commit/c2eddb433ab1014933531ef80057d0447d8dc86c

## Project

There is a semester long project for the class. It requires teams of three people and is meant to address something currently occurring in data center research. There are three milestones for the project. The first is before spring break, the second midway between spring break and the end of the semester, with the last milestone in the final week of classes.
The project is not meant to be time intensive, however it does contribute the most to your final course grade. The professors and TAs are happy to help you come up with an interesting topic. 

> Your team will have to write a report for each milestone to show progress. 

Reviewer: How long are these reports? Are they graded harshly?

Author: See commit https://github.com/gfieni2022/CMU-ECE-CS-Guide/commit/c2eddb433ab1014933531ef80057d0447d8dc86c

## Pros of the Class

- I greatly enjoyed being exposed to new topics each week.
- Do not have to be familiar with every topic covered.
- Class discussions are very interesting and you learn a lot during them.

> - TAs and professors are very approachable and easy to contact, since the class has a slack workspace.

Reviewer: ```suggestion
- TAs and professors are very approachable and easy to contact, since the class has a Slack workspace.
```

## Cons of the Class

- Fast paced, difficult to ingest new information so frequently.
- Not enough time is given to the project in class.

> - Reviews often take up much time in the week. 

Reviewer: ```suggestion
- Reviews often take up a lot of time in the week. 
```

## Resources

> - [Github for the class] https://github.com/STAR-Research/18-847C-S22

Reviewer: ```suggestion
- [Github for the class](https://github.com/STAR-Research/18-847C-S22)
```
